diff --git src/core/ngx_syslog.c src/core/ngx_syslog.c
index d4e79f6..f4a2703 100644
--- a/src/core/ngx_syslog.c
+++ b/src/core/ngx_syslog.c
@@ -182,7 +182,7 @@ ngx_syslog_parse_args(ngx_conf_t *cf, ngx_syslog_peer_t *peer)
             for (i = 4; i < len; i++) {
                 c = ngx_tolower(p[i]);

-                if (c < '0' || (c > '9' && c < 'a' && c != '_') || c > 'z') {
+                if ((c < '0' && c != '-') || (c > '9' && c < 'a' && c != '_') || c > 'z') {
                     ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
                                        "syslog \"tag\" only allows "
                                        "alphanumeric characters "
